traffic_ops/app/lib/API/DeliveryService/KeysUrlSig.pm

```diff
@@ -69,14 +86,17 @@ sub copy_url_sig_keys {
 	else {
 		return $self->alert("Delivery Service to copy from '$copy_from_xml_id' does not exist.");
 	}
+	if (!$tenant_utils->is_ds_resource_accessible($tenants_data, $copy_rs->tenant_id)) {
+		return $self->forbidden("Forbidden. Source delivery-service tenant is not available to the user.");
+	}
 	my $copy_config_file = $self->url_sig_config_file_name($copy_from_xml_id);
 
 	my $helper = new Utils::Helper( { mojo => $self } );
 	my $url_sig_key_values_json;
 
 	#verify we can copy keys out
 	if ( $helper->is_valid_delivery_service($copy_ds_id) ) {
-		if ( $is_admin || $helper->is_delivery_service_assigned($copy_ds_id) ) {
+		if ( $is_admin || $helper->is_delivery_service_assigned($copy_ds_id) || $tenant_utils->use_tenancy()) {
```

Comment:
I don't understand this line. If use_tenancy is 1 then it doesn't matter if you are an admin or you have the ds assigned you simply have to have the right tenant.....right?